### PR TITLE
IOGX Integration

### PR DESCRIPTION
**Pull request by zeme-wana**
As part of the ongoing effort to build a Developer Platform for IOG, the smart-contracts-dev-empowerment team is releasing [`IOGX - a Flake Template for Haskell Projects at IOG`](https://github.com/input-output-hk/iogx)

_The vision of IOGX is to provide a JSON-like, declarative interface to Nix, so that developers unfamiliar with the language may independently maintain and add to the Nix sources with minimum effort and maximum pleasure._

The goal is to integrate with all Smart Contracts repositories, and maybe more. 

There are currently 3 repositories using this: `marlowe-cardano`, `antaeus`, `quickcheck-dynamic`.

This PR deletes all legacy nix code and integrates with IOGX, for a reduction of ~86% of nix lines of code (1153 -> 165).

As developers nothing changes for you: you will enter `nix develop` and you will get into a new shell.

There might be some rough edges but we will be prompt to smooth them. 

--- 

Also note that while `fourmolu` and `shellcheck` support has been added, the hooks have been disabled temporarily in `nix/pre-commit-check.nix`.

After this PR is merged, someone will have to enable the hooks in `nix/pre-commit-check.nix`, and then while inside `nix develop`, run:
```
pre-commit run --all-files
``` 
Apply all the change and then open a new PR with the formatting changes. 
Basically this is to keep both PRs separate and clean

### CODE REVIEW

**Comment by zeme-wana** [comment]
`ci/eval` is pending but everything is actually green: https://ci.iog.io/eval/2471
It's a hydra thing

**Comment by zeme-wana** [comment]
CI is getting confused again. Everything passes, even cross-builds. This can be reviewed

**Review by kayvank — COMMENTED** [review]
LGT

**Review by kayvank — APPROVED** [review]
LGTM

File: .hindent.yaml [hunk 1]
```diff
@@ -1,3 +0,0 @@
-indent-size: 4
```
[the remark this answers is not in the document]
Comment (zeme-wana) [reply .hindent.yaml:1]: I deleted this file because I thought you wanted to move to fourmolu (see this PR's description message)

**Comment by zeme-wana** [comment]
Also, I deleted `.github/workflows/pre-commit-check.yml` because it's now run on Hydra by default

File: CONTRIBUTING.adoc [hunk 2]
```diff
@@ -329,9 +302,11 @@ package repository
...
 
 Getting new packages from there works much like getting them from Hackage.
-The differences are that it can have an independent ``index-state``, and
-that there is a different Nix command you need to run afterwards: ``nix flake
-lock --update-input CHaP``.
+The differences are that it can have an independent ``index-state``. 
+Refer to `this link
+<https://github.com/input-output-hk/iogx#312-inputs>`in IOGX's documentation 
+to learn how to update ``CHaP``.
+
```
Comment (koslambrou) [inline CONTRIBUTING.adoc:309]: Like in `hackage-nix`, maybe mention here how to update CHaP?

```
You can do this by running ``nix flake lock --update-input CHaP`.
```

File: nix/haskell-project.nix [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+# https://www.github.com/input-output-hk/iogx#33-nixhaskell-projectnix
+
+{ inputs, inputs', meta, config, pkgs, lib }:
+
+let
+
+  lib = pkgs.lib;
+
+
+  isCross = pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform;
```
Comment (koslambrou) [inline nix/haskell-project.nix:11]: Add comment on why this is necessary

File: nix/read-the-docs.nix [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# This file is part of the IOGX template and is documented at the link below:
+# https://www.github.com/input-output-hk/iogx#37-nixread-the-docsnix
+
+{
+  siteFolder = null; #"doc/read-the-docs-site";
```
Comment (koslambrou) [inline nix/read-the-docs.nix:5]: What's wrong with this? Why null?

File: CONTRIBUTING.adoc [hunk 5]
```diff
@@ -135,22 +135,9 @@ We don't have a `hie.yaml`, the implicit cradle support in HLS seems to work fin
 [[build-with-nix]]
 === How to build the project's artifacts with Nix
 
-Haskell components are provisioned by Nix via link:https://github.com/input-output-hk/haskell.nix[Haskell.nix]
+Haskell components are provisioned by Nix through link:https://github.com/input-output-hk/iogx[IOGX] via link:https://github.com/input-output-hk/haskell.nix[Haskell.nix]
 
-In general you can run `nix build .#SYSTEM.marconi.library.marconi-project.hsPkgs.PACKAGE.components.COMPONENT`,
-where `SYSTEM` is either `x86_64-linux` or `x86_64-darwin`.
-
-For example `nix build .#x86_64-linux.marconi.library.marconi-project.hsPkgs.marconi-sidechain.components.library` to build marconi-sidechain.
```
[the remark this answers is not in the document]
Comment (zeme-wana) [reply CONTRIBUTING.adoc:143]: The format of derivation names has changed.
It's now:

`$package-{exe|test|lib|bench|sublib}-$component-$ghc`
E.g. 
```
nix build .#marconi-sidechain-lib-marconi-sidechain-ghc8107
```

File: nix/haskell-project.nix (hunk 3 above)
Comment (zeme-wana) [reply nix/haskell-project.nix:11]: Done

File: nix/read-the-docs.nix (hunk 4 above)
Comment (zeme-wana) [reply nix/read-the-docs.nix:5]: Oversight. Fixed now.

**Comment by zeme-wana** [comment]
We do have one failing test now https://ci.iog.io/build/231172/nixlog/1

File: CONTRIBUTING.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (koslambrou) [reply CONTRIBUTING.adoc:143]: Is that documented somewhere? Seems helpful to know.

**Comment by koslambrou** [comment]
@zeme-wana I believe that's a fiddle test case. You can force push again to see if it passes?

File: CONTRIBUTING.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (zeme-wana) [reply CONTRIBUTING.adoc:143]: | Once inside the `nix develop` shell, you can run the `list-nix-outputs` script to list all buildable Nix derivations.
 
I've replaced line 140 with this ^^^^
Also here https://github.com/input-output-hk/iogx#310-flake-outputs-format

[the remark this answers is not in the document]
Comment (zeme-wana) [reply CONTRIBUTING.adoc:143]: | Once inside the `nix develop` shell, you can run the `list-nix-outputs` script to list all buildable Nix derivations.
 
I've replaced line 140 with this ^^^^
Also here https://github.com/input-output-hk/iogx#310-flake-outputs-format

**Comment by zeme-wana** [comment]
Yes all tests pass now, though CI is still confused

File: CONTRIBUTING.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (koslambrou) [reply CONTRIBUTING.adoc:143]: Nice! FYI, your IOGX link doesn't include the `list-nix-outputs` command

[the remark this answers is not in the document]
Comment (zeme-wana) [reply CONTRIBUTING.adoc:143]: thanks, oversight on my part. it does now
